### PR TITLE
Confirm approval

### PR DESCRIPTION
**Pull request by butofleury**
lock-unlock-confirmation

### CODE REVIEW

File: src/components/invoices/InvoiceToolBar.js [hunk 1]
```diff
@@ -284,6 +261,28 @@ class InvoiceToolBar extends Component {
...
+        <Dialog
+          open={this.state.locked}
+          onClose={this.handleUnLock}
+          aria-labelledby="alert-dialog-title"
+          aria-describedby="alert-dialog-description"
+        >
+          <DialogTitle id="alert-dialog-title">{actionState}</DialogTitle>
+          <DialogContent>
+            <DialogContentText id="alert-dialog-description">
+              This will {actionState} Data for the Zone {this.props.invoice.orgUnit && this.props.invoice.orgUnit.name}{" "}
```
Comment (mestachs) [inline src/components/invoices/InvoiceToolBar.js:273]: may be say parent and its orgunits (not always called a zone)
Ideally I would have displayed the don't know how hard it is to turn the dataApproval.orgUnit into a list of orgunits.
I'll try to test that this afternoon.

**Comment by butofleury** [comment]
After clicked on the Lock/Unlock button, a dialog with a message is open to confirm the action:


image

File: src/components/invoices/InvoiceToolBar.js (hunk 1 above)
Comment (butofleury) [reply src/components/invoices/InvoiceToolBar.js:273]: Capture d’écran de 2021-03-12 11-18-09

**Comment by mestachs** [comment]
Sorry but we need "both" lock and unlock button, here I didn't noticed but you replaced both by a single button.

image

Exemple you can lock a district "alone" and then check the "approvals" report, and get something as below. 

image

I want to be able
- to unlock the "District Ouelessebougou" or 
- lock all the remaining districts



and the display of the "zones" should the one from the dataApprovals (it's not necessarily the orgUnits of the invoice, eg in your screenshot it was schools... but we lock "zones")

I would extract a component and passing "existing functions". 
We can start this tomorrow together and you finish on your own.

**Comment by butofleury** [comment]
Some improvements :

image


image

**Pull request merged**